pkg/operator/ceph/webhook.go

```diff
@@ -30,35 +29,70 @@ import (
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/apimachinery/pkg/util/version"
 	"sigs.k8s.io/controller-runtime/pkg/webhook"
 )
 
 const (
 	admissionControllerAppName       = "rook-ceph-admission-controller"
 	tlsPort                    int32 = 443
+	minK8sVersionWebhook             = "1.16.0"
 )
 
 var (
-	namespace = os.Getenv(k8sutil.PodNamespaceEnvVar)
+	namespace              = os.Getenv(k8sutil.PodNamespaceEnvVar)
+	certManagerWebhookName = "cert-manager-webhook"
 )
 
 func isSecretPresent(ctx context.Context, context *clusterd.Context) (bool, error) {
+	k8sVersion, err := k8sutil.GetK8SVersion(context.Clientset)
+	if err != nil {
```

Comment:
This method either needs to be renamed, or it needs to be split up. It does a lot more than check if a secret exists. :)

Reply:
renamed